doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,674 @@
+FABRID also implicitly implements the `EPIC' <https://netsec.ethz.ch/publications/papers/Legner_Usenix2020_EPIC.pdf>`_
+features, including source authentication for routers and the destination host, and path validation for source and destination hosts.
+
+Proposal
+========
+
+FABRID indroduces policies, which can be thought of as additional path constraints: ASes on an endhost's selected inter-domain path only
+forward traffic over intra-AS paths of their network that match the endhost's selected policy.
+ASes announce the policies they satisfy to endhosts, which use them during path selection to further constrain the desired forwarding paths.
+The endhosts communicate the selected policies to the on-path ASes by encoding them in SCION data packets.
+On-path border routers read the encoded policies from each packet, and forward the packet along one of potentially multiple policy-compliant intra-AS paths.
+Enforcing correct traversal of an intra-AS path can for example be achieved through MPLS.
+Some FABRID policies are globally defined and others locally per AS.
+Global policies serve the purpose of avoiding redundancy for policies implemented by many ASes.
+The AS network operator specifies in a configuration which global policies are supported, and can additionally define arbitrary local policies.
+Since each AS can create their own local policies, endhosts need a mechanism to discover them.
+We distinguish between policy identifier and policy description.
+The policy identifiers are provided to the endhosts as a detachable PCB extension.
+The policy descriptions on the other hand have to be requested from the local control service which will then either return
+the cached policy description or query the policy description from the remote AS.
+By only encoding the policy identifiers, but not the policy descriptions, inside the PCBs, we can minimize FABRID's PCB size overhead, which thus becomes negligible.
+In our design, endhosts retrieve the policies from their local AS, and the local AS retrieves them from the remote AS, similar to how SCION path retrieval is implemented.
+Those policies are only fetched on demand by the local control service and will be cached until they expire.
+This on-demand policy retrieval improves the scalability of the control service compared to simply fetching all the policies from all ASes in the internet.
+A source endhost can then select a SCION forwarding path, where it encodes up to one policy index for each on-path AS (it can also select no policy for an AS) in its packets.
+On-path border routers will then add a proof of transit such that the destination endhost can later verify that the packet indeed followed the intended path.
+
+.. figure:: fig/FABRID/Overview.png
+
+    System overview. During beaconing, every AS adds its supported policies to the PCB (green). Endpoint EF fetches the
```

Comment:
Instead of "System overview" -> "Policy-driven intra-AS Routing with FABRID."

Reply:
EC, EF, PX, etc. should use an index (-> E_C, E_F, P_X, etc.)

Reply:
Done.